lib/checkmemoryleak.cpp

```diff
@@ -985,31 +985,40 @@ void CheckMemoryLeakNoVar::checkForUnreleasedInputArgument(const Scope *scope)
     // parse the executable scope until tok is reached...
     for (const Token *tok = scope->bodyStart; tok != scope->bodyEnd; tok = tok->next()) {
         // allocating memory in parameter for function call..
-        if (!(Token::Match(tok, "[(,] %name% (") && Token::Match(tok->linkAt(2), ") [,)]")))
-            continue;
-        if (getAllocationType(tok->next(), 0) == No)
-            continue;
-        // locate outer function call..
-        const Token* tok3 = tok;
-        while (tok3 && tok3->astParent() && tok3->str() == ",")
-            tok3 = tok3->astParent();
-        if (!tok3 || tok3->str() != "(")
+        if (!Token::Match(tok, "%name% ("))
             continue;
-        // Is it a function call..
-        if (!Token::Match(tok3->tokAt(-2), "!!= %name% ("))
+
+        // check if the output of the function is assigned
+        const Token* tok2 = tok->next()->astParent();
+        while (tok2 && tok2->isCast())
+            tok2 = tok2->astParent();
+        if (tok2 && tok2->isAssignmentOp())
             continue;
-        const std::string& functionName = tok3->strAt(-1);
+
+        const std::string& functionName = tok->str();
         if ((mTokenizer->isCPP() && functionName == "delete") ||
             functionName == "free" ||
             functionName == "fclose" ||
```

Comment:
This should use the library configurations instead, and I do think it should be `continue` instead of `break`, but one thing at a time.